### PR TITLE
core/types: copy tx recipient address

### PR DESCRIPTION
**Pull request by aaronbuchwald**
This PR creates a helper function `CopyAddress` to copy address pointers and uses it to address an outstanding TODO related to copying To addresses in transactions.

### CODE REVIEW

**Comment by aaronbuchwald** [comment]
Wondering if it's cleaner to add a function `common.CopyAddress(addr)` instead

File: common/types.go [hunk 1]
```diff
@@ -339,6 +339,16 @@ func (a Address) Value() (driver.Value, error) {
 	return a[:], nil
 }
 
+// CopyPointer returns a new pointer to a copy of the address or nil
+// if [a] is nil.
+func (a *Address) CopyPointer() *Address {
```
Comment (holiman) [inline common/types.go:344]: I think I'd prefer it called `Clone` personally. And the doc could be a bit clearer. 
" Clone returns a pointer to a copy of a (or nil if a is nil)"

Just a suggestion, really

Comment (aaronbuchwald) [reply common/types.go:344]: `Clone` sounds much better to me. I'll amend this today.

**Comment by fjl** [comment]
We have discussed this and would rather have the function to copy an address pointer in core/types. The method is too specialized to be useful.

**Pull request merged**